include/wabt/type.h

```diff
@@ -60,19 +60,40 @@ class Type {
 
   Type() = default;  // Provided so Type can be member of a union.
   Type(int32_t code)
-      : enum_(static_cast<Enum>(code)), type_index_(kInvalidIndex) {}
-  Type(Enum e) : enum_(e), type_index_(kInvalidIndex) {}
+      : enum_(static_cast<Enum>(code)), type_index_(0) {
+    assert(!EnumIsReferenceWithIndex(enum_));
+  }
+  Type(Enum e) : enum_(e), type_index_(0) {
+    assert(!EnumIsReferenceWithIndex(enum_));
+  }
   Type(Enum e, Index type_index) : enum_(e), type_index_(type_index) {
-    assert(e == Enum::Reference);
+    assert(EnumIsReferenceWithIndex(e));
   }
   constexpr operator Enum() const { return enum_; }
 
+  friend constexpr bool operator==(const Type a, const Type b) {
```

Comment:
Is Type or Type& is better?

Reply:
Type is probably better, it's a pair of 32-bit values after all. Any modern compiler should be able to optimize it.